src/feedbacks/loading/spinner/PSpinner.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+    <svg class="p-spinner"
+         :class="[styleType, size]"
+         viewBox="0 0 32 32"
+         fill="none" xmlns="http://www.w3.org/2000/svg"
+    >
+        <g>
+            <circle class="spinner-bg"
+                    cx="16" cy="16" r="14"
+                    stroke-opacity="0.3"
+                    stroke-width="9%"
+            />
+            <circle class="spinner"
+                    cx="16" cy="16" r="14"
+                    stroke-width="9%"
+                    stroke-linecap="round" stroke-dasharray="32 200"
+            />
+        </g>
+    </svg>
+</template>
+
+<script lang="ts">
+import {
+    defineComponent,
+} from 'vue';
+
+import { SPINNER_SIZE, SPINNER_STYLE_TYPE } from '@/feedbacks/loading/spinner/type';
+
+interface Props {
+    size?: string;
+    styleType?: string;
+}
+
+export default defineComponent<Props>({
+    name: 'PSpinner',
+    components: {
+    },
+    props: {
+        size: {
+            type: String,
+            default: SPINNER_SIZE.md,
+        },
+        styleType: {
+            type: String,
```

Comment:
I think it would be better with PropType. How about this ?
```suggestion
            type: String as PropType<keyof typeof SPINNER_STYLE_TYPE>,
```